fabtests/common/shared.c

```diff
@@ -578,18 +586,13 @@ int ft_alloc_msgs(void)
 		if (ret)
 			return ret;
 	}
+
 	ret = ft_hmem_memset(opts.iface, opts.device, (void *) buf, 0, buf_size);
 	if (ret)
 		return ret;
-	rx_buf = buf;
-
-	if (opts.options & FT_OPT_ALLOC_MULT_MR)
-		tx_buf = (char *) buf + MAX(rx_size, FT_MAX_CTRL_MSG);
-	else
-		tx_buf = (char *) buf +
-			 MAX(rx_size, FT_MAX_CTRL_MSG) * opts.window_size +
-			 rma_resv_bytes;
 
+	rx_buf = (char *)(((uintptr_t)buf + alignment - 1) / alignment * alignment);
```

Comment:
This doesn't look right.  Dividing by alignment, then multiplying by alignment is basically multiplying by 1.  It's also non-trivial why the rx_buf has some weird adjustment, but tx_buf below does not.  Maybe tx_buf should have been rx_buf + half_buf_size?

I think we need some macro here to help with readability as to the intent.

Reply:
I think you want something like this:
```
rx_buf = ((uintptr_t) buf + alignment - 1) & ~((uintptr_t) (alignment - 1));
```
The right hand side is an aligned buffer given an input pointer and could be a macro.  I think this is close to the C++ align call, but don't know if that's available for us anywhere.  I didn't see a similar macro in the libfabric source.

Reply:
Cool, I added a function `ft_get_aligned_addr()` which wrap up your code.